File: 504-base-7/base-7.java
class Solution {
    public String convertToBase7(int num) {
        if (num == 0) return "0";  

        boolean neg = num < 0;     
        int n = Math.abs(num);     
        StringBuilder sb = new StringBuilder();
        
        while (n > 0) {            
            sb.append(n % 7);     
            n /= 7;               
        }

        String res = sb.reverse().toString(); 
        return neg ? "-" + res : res;        
    }
}

File: 504-base-7/base-7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"
        neg = num < 0
        n = abs(num)
        a = ""
        while n > 0:
            a = str(n % 7) + a
            n //= 7
        return "-" + a if neg else a
